docusaurus-docs/community/policies/backwards-compatibility.md

```diff
@@ -139,42 +139,92 @@ get community feedback.
-return a status code of `410 Gone` to indicate the endpoint has been removed
-along with a json object stating when and why the endpoint was removed and what
+The `Sunset` date indicates the date after which conda store with remove the functionality
+of this endpoint.
+* Any time before this date, users should expect this endpoint to work
+* Any time after this date, the endpoint may still be available (due to no release having gone out), but users should expect that this endpoint will be removed
+
+Since the conda-store project users CalVer, the `Sunset` date should be set to at least 2 months
+from the release that the deprecation notice first appears in.
+
+##### 2. Remove endpoint functionality
+
+Once we have reached the `Sunset` date (from the step above), conda-store may
+remove the functionality of the endpoint. To indicate that this endpoint is no
+longer functional, the endpoint must:
+* return a status code of `410 Gone`
+* return a json object stating when and why the endpoint was removed and what
 version of the endpoint is available currently (if any).
 
+For example:
+
 ```python
 {
   # the pull request that removed the endpoint
   "reference_pull_request": "https://github.com/conda-incubator/conda-store/pull/0000",
-  # the date the endpoint was removed
-  "removal_date": "2021-06-24",
+  # the date the endpoint was removed (as HTTP-Date). This should be after the Sunset date specified
+  # in the deprecation step
+  "removal_date": "Mon, 16 Feb 2025 23:59:59 UTC"
   # the reason for the removal, ideally with a link to a CVE if one is available
   "removal_reason": "Removed to address CVE-2021-32677 (https://nvd.nist.gov/vuln/detail/CVE-2021-32677)",
-  # the endpoint that developers should use as a replacement
+  # the endpoint that developers should use as a replacement (if applicable)
   "new_endpoint": "api/v3/this/should/be/used/instead",
 }
 ```
 
-If an API endpoint must be deprecated, a deprecation warning should be added
-for at least one release before the endpoint is removed. This requirement may
-be waived in the case of a serious security vulnerability.
+###### Choosing a removal date
 
-It should always be clearly communicated in release notes and documentation
-when an API endpoint is deprecated or removed. This should include:
-- version number of the release where this was deprecated
-- provide suggestions for alternatives (if possible)
-- provide justification for the removal (such as a link to the issue
-  or CVE that necessitated the removal).
+The `removal_date` date indicates the date after which conda store with remove the endpoint
+from the codebase. After this point, the endpoint will return a `404` response code.
+* Any time before this date, users should expect this endpoint to return a `410` response code
+* Any time after this date, the endpoint may still return a `410` response (due to no release having gone out), but users should expect that this endpoint will disappear
```

Comment:
I am having trouble following these points. Per the above (line 190)

> To indicate that this endpoint is no
longer functional, the endpoint must:
* return a status code of `410 Gone`
* return a json object stating when and why the endpoint was removed and what
version of the endpoint is available currently (if any).

Once we reach the `sunset` (or removal date), the endpoint should return a `410`, but the bullets here seem to indicate that the endpoint should return a `410` _before_ the `sunset` date and a `404` after, which contradicts the steps above.

These two sections need aligning. To keep things simple we should just stick to a `410` I think,